File: BankatiProject_Backend/services/User/src/main/java/com/example/user/users/dto/UserDto.java
package com.example.user.users.dto;

import com.example.user.users.entity.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserDto {

    private String id;
    private String agentId;
    private String firstName;
    private String lastName;
    private String email;
    private String address;
    private String cin;
    private Date birthDate ;
    private String phoneNumber;
    private Role role ;
    private String password;
    private boolean isFirstLogin ;
    private Date createdDate;
    private String commercialRn;
    private String image;
    private String patentNumber ;
    private Boolean isPaymentAccountActivated ;
    private String typeHissab;
    private String currency;
    private String token;
    private String message;


}

File: BankatiProject_Front-end/src/app/models/UserDto.ts
// user-dto.model.ts
export interface UserDto {
    id: string;
    agentId: string;
    firstName: string;
    lastName: string;
    email: string;
    address: string;
    cin: string;
    birthDate: Date;
    phoneNumber: string;
    role: string; // Vous pouvez soit utiliser `string` pour le rôle ou un modèle de rôle si vous en avez un
    password: string;
    isFirstLogin: boolean;
    createdDate: Date;
    commercialRn: string;
    image: string;
    patentNumber: string;
    isPaymentAccountActivated: boolean;
    typeHissab: string;
    currency: string;
    token: string;
    message: string;
}
